app/controllers/application_controller.rb

```diff
@@ -22,11 +18,15 @@ def shib_flash
   end
 
   def shib_sign_up_redirect
-    if !user_signed_in? && @aaf_credentials[:email].present? && User.find_by_email(@aaf_credentials[:email]).blank?
+    if !user_signed_in? && @aaf_credentials[:mail].present? && User.find_for_authentication(email: @aaf_credentials[:mail]).blank?
```

Comment:
References to @aaf_credentials[:mail] is duplicate in a number of places which makes it harder to maintain. It would be better to move all the AAF authentication code into a separate module then include it as a helper module into the controller then you can reference it in the controller and the view.